### DISCUSSION TITLE
no uasyncio module present on ESP8266

### DISCUSSION THREAD

**Discussion by dresdner353** [micropython/micropython#9350] [Hardware & Peripherals] [answered]
Folks,
   I've installed the latest 1Mb stable ESP8266 version of micropython (esp8266-1m-20220618-v1.19.1.bin) on one of my boards and later tried a more recent nightly build. 

The issue I'm seeing is that neither image appears to have the uasyncio module present.

```
>>> help('modules')
__main__          gc                ucollections      urandom
_boot             inisetup          ucryptolib        ure
_onewire          lwip              uctypes           uselect
_webrepl          math              uerrno            usocket
apa102            micropython       uhashlib          ussl
btree             neopixel          uheapq            ustruct
builtins          network           uio               usys
dht               ntptime           ujson             utime
ds18x20           onewire           umachine          uwebsocket
esp               port_diag         uos               uzlib
flashbdev         uarray            upip              webrepl
framebuf          ubinascii         upip_utarfile     webrepl_setup
Plus any modules on the filesystem
>>> import uasyncio
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: no module named 'uasyncio'
```

I've failed to find any exact confirmation if its just not supported in the ESP8266 port. The reason I need this module is that I want to leverage Microdot as a web server layer and it requires the uasyncio module.

Any ideas?

**Comment by scruss** [comment, marked as answer]
Yes, there isn't enough room to include uasyncio on the 1 MB flash boards. They're long obsolete

**Reply by dpgeorge** [reply]
That's correct, uasyncio is not included by default on the 1MB board.

You could try freezing it in yourself, and removing other things you don't need, eg dht, onewire, neopixel, webrepl, upip.

You'd also probably need to freeze microdot.

This is the file that describes what's frozen into the 1MB build: https://github.com/micropython/micropython/blob/master/ports/esp8266/boards/manifest.py
